orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/manifest/WaitForManifestStableTask.java

```diff
@@ -55,6 +57,20 @@ public long getTimeout() {
     return TimeUnit.MINUTES.toMillis(30);
   }
 
+  @Override
+  public long getDynamicTimeout(StageExecution stage) {
+    ManifestTimeout timeout = stage.mapTo(ManifestTimeout.class);
+    if (timeout.timeoutInMinutes != null) {
```

Comment:
This seems to be mixing having `timeoutInMinutes` be `@Nullable` and having it be an `Optional`; you're checking that the `Optional<Long>` is not `null` but never checking `.isPresent()` before calling `.get()` which will throw an exception if it's an empty `Optional`.